integrated_channels/sap_success_factors/models.py

```diff
@@ -93,9 +114,41 @@ class LearnerDataTransmissionAudit(models.Model):
     error_message = models.TextField(blank=True)
     created = models.DateTimeField(auto_now_add=True)
 
+    epoch = datetime(1970, 1, 1, tzinfo=timezone.utc)
+    incomplete_grade = 'incomplete'
+
     class Meta:
         app_label = 'sap_success_factors'
 
+    def __init__(self, *args, **kwargs):
+        """
+        Return a new ``LearnerDataTransmissionAudit`` object.
+
+        Optional arguments:
+        * ``enterprise_enrollment``: If provided, then ``enterprise_course_enrollment_id``, ``user_id``, * ``course_id``
+          are pulled from this EnterpriseCourseEnrollment object.
+        * ``certificate``: If provided, then ``course_completed=True``; ``completed_timestamp`, ``grade``, ``status``
+          are pulled from this GeneratedCertificate object.
+
+        Otherwise, the field values are pulled directly from the kwargs, as usual.
+        """
+        enterprise_enrollment = kwargs.pop('enterprise_enrollment', None)
+        if enterprise_enrollment is not None:
+            kwargs['enterprise_course_enrollment_id'] = enterprise_enrollment.id
+            # TODO: ENT-164 will provide a way to get the SAP User ID for this user from the SAP SSO backend.
```

Comment:
I think [ENT-190](https://openedx.atlassian.net/browse/ENT-190) might be the closest story?  I didn't want to pin it to that, in case it required extra work beyond basic SSO.